### PR TITLE
[CAP] Abstraction of actor_connector to go along with runtime factory and runtime abstraction

### PR DESCRIPTION
**Pull request by rajan-chari**
## Why are these changes needed?

AutoGen CAP comes with a default runtime, that's ready to use out of the box, to make it easy to start building Agents in minutes.  However, different use cases and environments may require their own runtimes to host AutoGen CAP Agents.  Runtime factories were added to make it easy to create new runtimes.  

Along with runtime factories runtimes were also abstracted so that Agents/Actors can switch to different runtime implementations as needed.  

This PR introduces and abstraction for agent connectors since they will depend on the needs of each runtime.  Default implementation is also provided to make it super easy to start building Agents with AutoGen CAP.

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

File: samples/apps/cap/py/autogencap/ag_adapter/AG2CAP.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from autogen import Agent, ConversableAgent
 
 from ..actor_runtime import IRuntime
-from .AutoGenConnector import AutoGenConnector
+from .autogen_connector import AutoGenConnector
 
 
 class AG2CAP(ConversableAgent):
```
Comment (thinkall) [inline samples/apps/cap/py/autogencap/ag_adapter/AG2CAP.py:10]: Would you like to also rename this file to something like "ag2cap.py"?

**Review by thinkall — COMMENTED** [review]
Thanks for the PR, @rajan-chari . I saw you renamed some filenames from upper cases to lower cases, would you like to rename others?

File: samples/apps/cap/py/autogencap/ag_adapter/AG2CAP.py (hunk 1 above)
Comment (rajanjedi) [reply samples/apps/cap/py/autogencap/ag_adapter/AG2CAP.py:10]: Yes.  Makes sense.  Thanks.

Comment (rajan-chari) [reply samples/apps/cap/py/autogencap/ag_adapter/AG2CAP.py:10]: I am reconsidering some abstractions for the actor itself.  Converting this to draft.  Thanks @thinkall

File: samples/apps/cap/py/autogencap/actor_connector.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Agent_Sender takes a zmq context, Topic and creates a
+# socket that can publish to that topic. It exposes this functionality
+# using send_msg method
+from abc import ABC, abstractmethod
+
+
+class IActorSender(ABC):
```
Comment (heyitsaamir) [inline samples/apps/cap/py/autogencap/actor_connector.py:7]: It would be good to document each method and intention here

Comment (rajan-chari) [reply samples/apps/cap/py/autogencap/actor_connector.py:7]: Yeah good point.  Thanks.

**Comment by rysweet** [comment]
This PR is against AutoGen 0.2. AutoGen 0.2 has been moved to the 0.2 branch. Please rebase your PR on the 0.2 branch or update it to work with the new AutoGen 0.4 that is now in main.

**Comment by rysweet** [comment]
Hi Rajan,
we've rebased and updated the branch for you. If you are still considering bringing this PR forward please go ahead and update and mark as ready for review. You may also consider having a closer look at 0.4 which has moved to an event-driven actor model.  If you are not going to finish this PR please go ahed and close it.

**Comment by rajan-chari** [comment]
> Hi Rajan, we've rebased and updated the branch for you. If you are still considering bringing this PR forward please go ahead and update and mark as ready for review. You may also consider having a closer look at 0.4 which has moved to an event-driven actor model. If you are not going to finish this PR please go ahed and close it.

Hi Ryan,
I will update the PR and mark it as ready for review in a short while.

**Review by ekzhu — APPROVED** [review]

**Pull request merged**